### PR TITLE
SLI-1007 Resolve local-only issues when connected to SonarQube

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/actions/MarkAsResolvedAction.kt [hunk 1]
```diff
@@ -60,13 +61,18 @@ class MarkAsResolvedAction(
     private var issue: Issue? = null
 )  :
     AbstractSonarAction(
-        "Mark as Resolved", "Mark as Resolved", null
+        "Mark Issue as...", "Change the issue resolution status", null
     ), IntentionAction, PriorityAction, Iconable {
     companion object {
         private const val errorTitle = "<b>SonarLint - Unable to mark the issue as resolved</b>"
         private const val content = "The issue was successfully marked as resolved"
```
Comment (nquinquenel) [inline src/main/java/org/sonarlint/intellij/actions/MarkAsResolvedAction.kt:68]: Nitpick -> We could put the actual status in the message so the user has more feedback on his action

**Review by nquinquenel — APPROVED** [review]
LGTM, waiting to test with SQ PoC

**Review by nquinquenel — COMMENTED** [review]
When opening the context menu for an issue, I have this error:

`Error while retrieving the list of allowed statuses for issues
java.lang.IllegalStateException: Calling invokeAndWait from read-action leads to possible deadlock.
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:444)
	at com.intellij.openapi.application.ex.ApplicationUtil.invokeAndWaitSomewhere(ApplicationUtil.java:145)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:85)
	at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:367)
	at com.intellij.openapi.progress.ProgressManager.run(ProgressManager.java:207)
	at org.sonarlint.intellij.actions.MarkAsResolvedAction$Companion.checkPermission(MarkAsResolvedAction.kt:145)
	at org.sonarlint.intellij.actions.MarkAsResolvedAction$Companion.access$checkPermission(MarkAsResolvedAction.kt:66)
	at org.sonarlint.intellij.actions.MarkAsResolvedAction.isPermitted(MarkAsResolvedAction.kt:183)
	at org.sonarlint.intellij.actions.MarkAsResolvedAction.isEnabled(MarkAsResolvedAction.kt:177)
	at org.sonarlint.intellij.actions.AbstractSonarAction.update(AbstractSonarAction.java:61)`

When calling `ProgressManager.getInstance().run(checkTask)`

File: src/main/java/org/sonarlint/intellij/core/BackendService.kt [hunk 2]
```diff
@@ -400,18 +404,90 @@ class BackendService @NonInjectable constructor(private val backend: SonarLintBa
...
+            TrackWithServerIssuesParams(
+                moduleId(module),
+                rawIssuesByRelativePath,
+                shouldFetchIssuesFromServer
+            )
+        ).thenAccept { response ->
+            response.issuesByServerRelativePath.forEach { (serverRelativePath, trackedIssues) ->
+                val file = virtualFileByServerRelativePath[serverRelativePath] ?: return@forEach
+                val liveIssues = liveIssuesByFile[file] ?: return@forEach
+                liveIssues.zip(trackedIssues).forEach { (liveIssue, serverOrLocalIssue) ->
```
[the remark this answers is not in the document]
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarlint/intellij/core/BackendService.kt:468]: I fixed it, there was an error in the line number conversion

**Review by nquinquenel — APPROVED** [review]
LGTM

**Pull request merged**